### PR TITLE
naz/nft 2451 link devnet program to front

### PR DESCRIPTION
**Pull request by nazariyv**

### CODE REVIEW

File: src/lib/stores/web3auth.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    const { subscribe, set, update } = writable<Web3AuthState>({
+        provider: null,
+    });
+
+    return {
+        subscribe,
+        setProvider: (provider: Web3AuthState['provider']) => update(state => ({ ...state, provider })),
+        disconnect: () => set({ provider: null }),
+    };
+}
```
Comment (rombrom) [inline src/lib/stores/web3auth.ts:18]: Any reason we're not leveraging/adapting the store in `src/lib/user.ts` for this?

File: src/vendor/web3auth/rpc.ts [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+  PublicKey,
+  SystemProgram,
+  Transaction,
+  TransactionMessage,
+  VersionedTransaction
+} from '@solana/web3.js';
+import { type CustomChainConfig, type IProvider } from '@web3auth/base';
+import { SolanaWallet } from '@web3auth/solana-provider';
+
+// taken from: https://github.com/Web3Auth/web3auth-pnp-examples/blob/main/web-modal-sdk/blockchain-connection-examples/solana-modal-example/src/solanaRPC.ts#L252
```
Comment (rombrom) [inline src/vendor/web3auth/rpc.ts:13]: I'd need to look into this, but maybe this is why Solana has those wallet adapter packages so we don't have to write a complete RPC interface?

File: src/lib/stores/web3auth.ts (hunk 1 above)
Comment (nazariyv) [reply src/lib/stores/web3auth.ts:18]: wasn't aware of it. will move provider there

File: src/vendor/web3auth/rpc.ts (hunk 2 above)
Comment (nazariyv) [reply src/vendor/web3auth/rpc.ts:13]: looking at the code, I am not using it anywhere. might delete in the cleanup when I'm done with this PR.

**Pull request merged**